cloudferrylib/os/network/neutron.py

```diff
@@ -818,31 +818,74 @@ def upload_sec_group_rules(self, sec_groups):
     def upload_networks(self, networks):
         existing_nets_hashlist = (
             [ex_net['res_hash'] for ex_net in self.get_networks()])
+
+        # we need to handle duplicates in segmentation ids
+        used_seg_ids = [i.get('provider:segmentation_id') for i in networks]
+
+        def get_free_segmentation_id():
+            # this function finds first free seg_id
+            # adds it to array of used ids
+            # and returns seg_id
+            for i in xrange(1, 4096):
```

Comment:
> according to https://github.com/openstack/neutron/blob/stable/icehouse/neutron/plugins/ml2/drivers/type_tunnel.py#L77
GRE requires segmentation id (I get message NeutronClientException: 400-{u'NeutronError': {u'message': u'Invalid input for operation: segmentation_id required for gre provider network.', u'type': u'InvalidInput', u'detail': u''}})

This is only true for provider networks (when user wants to handle real physical network through Openstack network)

> Also - I don't like try except approach, because it will send more then one request per network, Also I don't think that we need that "number of retries", because we can lose data if we exceed number of retries. But I will find a way to handle different number of possible segmentation ids.

Only a suggestion on how to track segmentation IDs